google/cloud/bigquery/table.py

```diff
@@ -1726,7 +1727,32 @@ def to_dataframe(
                 bqstorage_client=bqstorage_client,
                 create_bqstorage_client=create_bqstorage_client,
             )
-            df = record_batch.to_pandas(date_as_object=date_as_object)
+
+            # When converting timestamp values to nanosecond precision, the result
+            # can be out of pyarrow bounds. To avoid the error when converting to
+            # Pandas, we set the timestamp_as_object parameter to True, if necessary.
+            #
+            # NOTE: Python 3+ only, as timestamp_as_object parameter is only supported
+            # in pyarrow>=1.0, but the latter is not compatible with Python 2.
+            if six.PY2:
+                extra_kwargs = {}
+            else:
+                type_to_check = pyarrow.timestamp("us", tz=pytz.UTC)
```

Comment:
I'm curious. Do BigQuery `DATETIME` columns include the timezone when coming back as Arrow (reminded by https://github.com/googleapis/python-bigquery/issues/201)

Reply:
It seems like they don't, thus I applied the same fix for no-timezone timestamps.